bus-mapping/src/evm.rs

```diff
@@ -58,125 +52,106 @@ impl From<usize> for GlobalCounter {
     }
 }
 
-/// Represents a `MemoryAddress` of the EVM.
+// XXX: Consider to move this to [u8;32] soon
+/// Representation of an EVM word which is basically a 32-byte word.
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
-pub struct MemoryAddress(pub(crate) BigUint);
+pub struct EvmWord(pub(crate) BigUint);
 
-impl MemoryAddress {
-    /// Returns the zero address for Memory targets.
-    pub fn zero() -> MemoryAddress {
-        MEM_ADDR_ZERO.clone()
+impl FromStr for EvmWord {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(EvmWord(
+            BigUint::from_str_radix(s, 16)
+                .map_err(|_| Error::EvmWordParsing)?,
+        ))
     }
+}
 
+impl_from_big_uint_wrappers!(
+    EvmWord = EvmWord,
+    (u8, u16, u32, u64, u128, usize)
+);
+
+impl EvmWord {
     /// Return the little-endian byte representation of the word as a 32-byte
     /// array.
     pub fn to_bytes(&self) -> [u8; 32] {
         let mut array = [0u8; 32];
-        array.copy_from_slice(&self.0.to_bytes_le());
+        let bytes = self.0.to_bytes_le();
+        array[..bytes.len() - 1].copy_from_slice(&bytes[0..bytes.len() - 1]);
```

Comment:
You're right.. Let me fix this..

Reply:
Maybe my question is why "N is always len-1". I tried to have a simple test which fails:

```rust
#[test]
fn test_evm_word() {
    let word_max = EvmWord::from_str("ff".repeat(32).as_str()).unwrap();
    assert_eq!(word_max.to_bytes(), [0xff; 32])
}
```

Or should they be different in our case?

Reply:
Ohh. It's always that because BigUint returns as a byte representation a vec with `len= #number of bytes`.
Therefore, since I want to export it to an array of 32 bytes, I need to generate the empty array and copy the bytes. And the amount of bytes will always be `len -1`. 

And just saying this I saw the error hahah. Sorry, my brain was AFK it seems. Fixing it!